packages/ove-service-imagetiles/src/OVE.Service.ImageTiles/Startup.cs

```diff
@@ -91,38 +94,58 @@ public class Startup {
         /// <summary>
         /// Register this OVE service with the Asset Manager Service 
         /// </summary>
-        /// <param name="app">kestrel app</param>
-        private async void RegisterServiceWithAssetManager(IApplicationBuilder app) {
+        private async void RegisterServiceWithAssetManager() {
+
+            // get the service description from the AppSettings.json 
+
             OVEService service = new OVEService();
             Configuration.Bind("Service", service);
-            
-            service.ViewIFrameUrl = Configuration.GetValue<string>("ServiceHostUrl") + "/api/ImageController/ViewImage/?id={id}"; 
 
-            // update the real processing states
+            service.ViewIFrameUrl = Configuration.GetValue<string>("ServiceHostUrl") +
+                                    "/api/ImageController/ViewImage/?id={id}";
+
+            // then update the real processing states
             service.ProcessingStates.Clear();
             foreach (var state in Enum.GetValues(typeof(ProcessingStates))) {
                 service.ProcessingStates.Add(((int) state).ToString(), state.ToString());
             }
 
             // register the service
 
-            string url = Configuration.GetValue<string>("AssetManagerHostUrl") +
-                         Configuration.GetValue<string>("RegistrationApi");
-
-            Console.WriteLine("About to register with url " + url + " we are on " + service.ViewIFrameUrl);
-
-            using (var client = new HttpClient()) {
-                var responseMessage = await client.PostAsJsonAsync(url, service);
-
-                Console.WriteLine("Result of Registration was " + responseMessage.StatusCode);
+            bool registered = false;
+            while (!registered) {
+                string url = null;
+                try {
+                    // permit environmental variables to be updated 
+                    url = Configuration.GetValue<string>("AssetManagerHostUrl") +
+                          Configuration.GetValue<string>("RegistrationApi");
+
+                    Console.WriteLine("About to register with url " + url + " we are on " + service.ViewIFrameUrl);
+
+                    using (var client = new HttpClient()) {
+                        var responseMessage = await client.PostAsJsonAsync(url, service);
+
+                        Console.WriteLine("Result of Registration was " + responseMessage.StatusCode);
+
+                        registered = responseMessage.StatusCode == HttpStatusCode.OK;
+                    }
+                }
+                catch (Exception e) {
+                    Console.WriteLine("Failed to register - exception was" + e);

```

Comment:
Don't we use any form of logging in these services? I've seen various logs written out by .NET having various types associated with it, I wouldn't expect the above to be written down as an error though. Can we easily fix this?

Reply:
Hi @senakafdo yes good point i figured you'd ask about this. 
What's going on here is that the logging system is not yet enabled at this stage of the OWIN pipeline. 
It's worth nothing that when running under docker all Console.WriteLine's are persisted to the docker logs so it will certainly be visible.

Reply:
There is a full logging system used throughout the rest of the services. Its just this initial configuration where the logger is not (yet) available

Reply:
I think it is available - https://stackoverflow.com/questions/41287648/how-do-i-write-logs-from-within-startup-cs ?

Reply:
Yes it is **now** possible to inject a logger at the start of this file. 
However this logger will not have been correctly configured at this time to redirect correctly. 

If you'd like me to inject a logger here please open a seperate issue and i will be happy to look into it. 
This is outside of the scope of this PR so am resolving this conversation.

Reply:
See #40.